store/src/java/com/zimbra/cs/store/external/ExternalBlobIO.java

```
@@ -38,6 +38,21 @@ public interface ExternalBlobIO {
      */
     String writeStreamToStore(InputStream in, long actualSize, Mailbox mbox) throws IOException, ServiceException;
 
+    /**
+     * Write data to blob store
+     *
+     * @param in:         InputStream containing data to be written
+     * @param actualSize: size of data in stream, or -1 if size is unknown. To be used by implementor for optimization where possible
+     * @param mbox:       Mailbox which contains the blob. Can optionally be used by store for partitioning
+     * @param destVolId:  Destination Volume id on which we need to store the blob
+     * @return locator string for the stored blob, unique identifier created by storage protocol
+     * @throws IOException
+     * @throws ServiceException
+     */
+    default String writeStreamToStore(InputStream in, long actualSize, Mailbox mbox, short destVolId) throws IOException, ServiceException {
+        return null;
+    }
```

Not able to understand why have we created the default method which is directly returning null?

---

@viditZim this being an interface is a way to fallback for methods to not require implementation for classes already implemented this interface to not break them.
@Gopal-Moolchandani pls check on the returning null.

---

You can throw exception such as `NotImplementedException` or `UnsupportedOperation`

---

@dasiyogesh Returning `null` from a default interface method is a bad design choice, regardless of backward compatibility concerns. This silently introduces `NullPointerExceptions` and violates the principle of least surprise.

The correct and safe approach is to throw an `UnsupportedOperationException` or similar instead of silently returning a `null` that could fail unpredictably at runtime. This is like introducing bug purposely and then fixing it.

---

right, it was returning null earlier thats what was to be checked and handle it for which above change is done.